lasagne/layers/conv.py

```diff
@@ -536,3 +606,176 @@ def convolve(self, input, **kwargs):
+        Initial value, expression or initializer for the weights.
+        These should be a 4D tensor with shape
+        ``(num_input_channels, num_filters, filter_rows, filter_columns)``.
+        Note that the first two dimensions are swapped compared to a
+        non-transposed convolution.
+        See :func:`lasagne.utils.create_param` for more information.
+
+    b : Theano shared variable, expression, numpy array, callable or ``None``
+        Initial value, expression or initializer for the biases. If set to
+        ``None``, the layer will have no biases. Otherwise, biases should be
+        a 1D array with shape ``(num_filters,)`` if `untied_biases` is set to
+        ``False``. If it is set to ``True``, its shape should be
+        ``(num_filters, output_rows, output_columns)`` instead.
+        See :func:`lasagne.utils.create_param` for more information.
+
+    nonlinearity : callable or None
+        The nonlinearity that is applied to the layer activations. If None
+        is provided, the layer will be linear.
+
+    flip_filters : bool (default: False)
+        Whether to flip the filters before sliding them over the input,
+        performing a convolution, or not to flip them and perform a
+        correlation (this is the default). Note that this flag is inverted
+        compared to a non-transposed convolution.
+
+    **kwargs
+        Any additional keyword arguments are passed to the `Layer` superclass.
+
+    Attributes
+    ----------
+    W : Theano shared variable or expression
+        Variable or expression representing the filter weights.
+
+    b : Theano shared variable or expression
+        Variable or expression representing the biases.
+
+    Notes
+    -----
+    The transposed convolution is implemented as the backward pass of a
+    corresponding non-transposed convolution. It can be thought of as dilating
+    the input (by adding ``stride - 1`` zeros between adjacent input elements),
+    padding it with ``filter_size - 1 - crop`` zeros, and cross-correlating it
+    with the filters. See [1]_ for more background.
+
+    Example
+    -------
+    To transpose an existing convolution, with tied filter weights:
+
+    >>> from lasagne.layers import Conv2DLayer, TransposedConv2DLayer
+    >>> conv = Conv2DLayer((None, 1, 32, 32), 16, 3, stride=2, pad=2)
+    >>> deconv = TransposedConv2DLayer(conv, conv.input_shape[1],
+    ...         conv.filter_size, stride=conv.stride, crop=conv.pad,
+    ...         W=conv.W, flip_filters=not conv.flip_filters)
+
+    References
+    ----------
+    .. [1]: Vincent Dumoulin, Francesco Visin (2016):
+            A guide to convolution arithmetic for deep learning. arXiv.
+            http://arxiv.org/abs/1603.07285,
+            https://github.com/vdumoulin/conv_arithmetic
+    """
+    def __init__(self, incoming, num_filters, filter_size, stride=(1, 1),
+                 crop=0, untie_biases=False,
```

Comment:
To be fair, the consistency isn't because I've paid extra close attention when writing the code, but because I started with copy/pasting the Conv2DLayer :)